stubs/PyYAML/yaml/nodes.pyi

```diff
@@ -1,23 +1,24 @@
+from _typeshed import Incomplete
 from typing import Any, ClassVar
 
 from yaml.error import Mark
 
 class Node:
     tag: str
     value: Any
-    start_mark: Mark | Any
-    end_mark: Mark | Any
+    start_mark: Mark | Incomplete
```

Comment:
Those attributes are directly assigned from the parameters, so they should be the same.
If it's possible, and intended to be anything else than `Mark | None`, with an arbitrary type, or if it's to avoid having to check for a None mark when you know you instantiated it not-None. And that's a common use-case/pattern. That can be represented using generics. Since it's feasible within the current typing system, `Incomplete` (could be done but we didn't bothered) seems preferable over `Any` (can't be done, we tried)

Reply:
I think there's an argument to be made here for either permissive unions or default generics. Which both aren't part of the type system atm. I've changed the type to `Mark | None | Any` and added a comment to explain.